### PR TITLE
Fix up reading auth schemes and setting default scheme

### LINKED ISSUE #42453: Set `DefaultScheme` in configuration from user-jwts tool

### LINKED ISSUE #42454: Fix reading scheme-specific options from config

### PR DESCRIPTION
**Pull request by captainsafia**
## Description

This PR resolves issues with reading scheme options from config by name and setting the default authentication scheme automatically when creating tokens using the`user-jwts` tool.

Fixes https://github.com/dotnet/aspnetcore/issues/42454 and fixes https://github.com/dotnet/aspnetcore/issues/42453

## Customer Impact

The changes in this PR address issues that make it difficult for users to get an end-to-end of our simplified authentication flow working in conjunction with the `user-jwts` tool.

## Regression?

- [ ] Yes
- [X] No

## Risk

- [ ] High
- [ ] Medium
- [X] Low

Changes only affect authentication area in ASP.NET Core and are limited to interactions with the `user-jwts` tool or the new `Authentication` property.

## Verification

- [ ] Manual (required)
- [X] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [X] N/A

### CODE REVIEW

File: src/Security/Authentication/test/AuthenticationMiddlewareTests.cs [hunk 1]
```diff
@@ -169,6 +175,9 @@ public async Task WebApplicationBuilder_RegistersAuthenticationMiddlewares()
         await app.StartAsync();
 
         Assert.True(app.Properties.ContainsKey("__AuthenticationMiddlewareSet"));
+
+        var options = app.Services.GetService<IOptionsMonitor<JwtBearerOptions>>().Get(JwtBearerDefaults.AuthenticationScheme);
+        Assert.Equal(new[] { "SomeIssuer" }, options.TokenValidationParameters.ValidIssuers);
```
Comment (BrennanConroy) [inline src/Security/Authentication/test/AuthenticationMiddlewareTests.cs:180]: Assert audience too?

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs [hunk 2]
```diff
@@ -92,7 +138,7 @@ public void Print_ReturnsNothingForMissingToken()
     public void List_ReturnsIdForGeneratedToken()
     {
         var project = Path.Combine(_fixture.CreateProject(), "TestProject.csproj");
-        var appsettings = Path.Combine(Path.GetDirectoryName(project), "appsettings.Development.json");
+        var appSettingsPath = Path.Combine(Path.GetDirectoryName(project), "appsettings.Development.json");
```
Comment (BrennanConroy) [inline src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:141]: remove?

File: src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs [hunk 3]
```diff
@@ -70,6 +80,11 @@ public static void RemoveScheme(string filePath, string name)
             authentication[SchemesKey] is JsonObject schemes)
         {
             schemes.Remove(name);
+            if (authentication[DefaultSchemeKey] is JsonValue defaultScheme
```
Comment (BrennanConroy) [inline src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs:83]: No tests for removing default scheme

Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/src/Helpers/JwtAuthenticationSchemeSettings.cs:83]: Good call. Updated the existing clear/remove tests to check for unsetting the DefaultScheme, also added a test to validate the DefaultScheme is not unset if the associated token isn't removed.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**